### DISCUSSION TITLE
Azioni da eseguire per catturare correttamente codici telecomando condizionatore tramite remote_learn_command

### DISCUSSION THREAD

**Discussion by federoco** [indomus/forum#3162] [HUB: Home Assistant]
Ciao comunità,
ho usato [remote_learn_command](https://indomus.it/guide/gestire-telecomandi-virtuali-su-home-assistant-via-broadlink-parte-1/) per acquisire i codici e poi creare vari [telecomandi virtuali](https://indomus.it/guide/gestire-telecomandi-virtuali-su-home-assistant-via-broadlink-parte-2/) per tv e soundbar e tutto funziona bene. 

Adesso vorrei fare la stessa cosa con i codici del condizionatore.

Vorrei usarli sia per creare un telecomando virtuale sia per inserirli in un .json personalizzato per SmartIR.
La teoria l'ho capita: la pressione di un tasto corrisponde sempre a una somma di parametri.
Non mi sono chiari i passaggi da eseguire nella pratica.

Mi spiego con un esempio.
Vorrei registrare le seguenti impostazioni dal telecomando (nb:  uso la terminologia che si usa nel file json)
operationModes: **dry**
fan modes: **auto**
swing modes: **low**
Chiamerò, d'ora in poi,  questo set di impostazioni che vorrei ottenere "_stato base_".

So che il climatizzatore, quando premo il tasto ON/OFF accende il condizionatore all'ultimo stato in cui l'ho spento.
Nel mio caso proprio le impostazioni dello "_stato base_".

Dunque come procedo per registrare questo stato?
Se parto con il telecomando del climatizzatore spento, basta che prema il tasto ON/OFF e che registri quello con remote_learn_command?

Oppure devo partire da uno "stato" diverso e registrare la pressione del tasto MODE del telecomando, che mi imposta di nuovo operationModes su **dry**?
Se si, allora la sequenza delle azioni sarebbe:
1) accendo il telecomando del condizionatore e imposto:
operationModes: **cool**
fan modes: auto
swing modes: low
temperature 26
2) metto in ascolto remote_learn_command su HA
3) premo il tasto MODE sul telecomando che imposta operationModes in "dry"
4) e ottengo la registrazione di tutte le impostazioni che corrispondono a "stato base".
E' corretto?

Intanto queste sono le prime perplessità. Grazie per il supporto

**Comment by indomus** [comment]
devi usare [SmartIR](https://indomus.it/progetti/domotizzare-un-condizionatore-tradizionale-con-broadlink-e-home-assistant-v2/).

**Reply by federoco** [reply]
Ciao, ti ringrazio ma uso già smartIR con successo ed ho integrato il condizionatore in HA, proprio seguendo la guida che mi hai indicato. 
Probabilmente mi sono spiegato male prima.

Quello che non mi è chiaro di quella guida è **la procedura da usare per catturare i codici tramite remote_learn_command**. 
Nella guida è scritto che, qualora non si trovi un file .json, allora è necessario crearlo a mano. E quindi catturare i codici.

Ripeto, il mio climatizzatore è integrato ma vorrei capire lo stesso come funziona il processo di cattura dei codici.
Non ho avuto problemi ad usare remote_learn_command per telecomandi di TV e soundbar ma non so come gestire il condizionatore.

**Reply by indomus** [reply]
devi usare https://indomus.it/guide/catturare-codici-infrarossi-e-radiofrequenza-usando-broadlink-manager-via-web-gui/

**Comment by federoco** [comment]
Grazie molte.
Bene. Ho studiato. Faccio un riassunto della teoria così come l'ho compresa per vedere se è tutto chiaro, prima di andare avanti con i miei test.
Puoi dirmi se ho compreso tutto quanto correttamente?

I climatizzatori vecchio stile con telecomandi IR si integrano e si comandano in HA tramite SmartIR.
Questo significa che SmartIR, per inviare al condizionatore i comandi utilizza lato hardware l'attuatore Broadlink e usa lato software il comando remote_send_command.
Fin qui è corretto quello che scrivo?

Inoltre remote_send_command ha 2 modalità di utilizzo.
Una è questa che ho appena descritto sopra per i condizionatori.

L'altra modalità di utilizzo di remote_send_command è per comandare apparecchi più semplici come una tv.
In quest'ultimo caso mi serve per creare telecomandi virtuali (argomento che avete affrontato in [un'altra guida](https://indomus.it/guide/gestire-telecomandi-virtuali-su-home-assistant-via-broadlink-parte-1/)) oppure posso chiamarlo direttamente - ad esempio in uno script -.

I comandi che posso inviare con questa modalità sono quelli che, a differenza del condizionatore, ho registrato con remote_learn_command da _Strumenti per sviluppatori --> Servizi_ e che si trovano nel file /.storage/broadlink_remote_000000000_codes
Anche qui ci siamo? Ho compreso bene?

**Reply by KC-inDomus** [reply]
No, anche nel caso delle TV si usa SmartIR. Cmq sì, se usi learn_command i codici acquisiti finiscono lì dove hai indicato.
